sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/RuleExecutor.scala

```diff
@@ -169,7 +169,7 @@ abstract class RuleExecutor[TreeType <: TreeNode[_]] extends Logging {
            |Once strategy's idempotence is broken for batch ${batch.name}
            |${sideBySide(plan.treeString, reOptimized.treeString).mkString("\n")}
           """.stripMargin
-      throw new TreeNodeException(reOptimized, message, null)
+      throw new AnalysisException(message)
```

Comment:
Is this always an analysis exception? Seems to me it is not only for analyzer rule.

Reply:
For a place like this, should we keep tree node exception?

Reply:
`TreeNodeException` makes stacktrace verbose and complicated. Yeah! it was work for analyzer, optimizer and expressions, do we need to create `RuleExeception`? or you have another idea.

Reply:
`RuntimeException` is also fine. It's test-only and most likely people don't care about exception type here.

Reply:
sounds okay.

Reply:
OK

Reply:
sgtm, too.